challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map(&:capitalize)
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.find_all {|animal| animal == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject {|animal| animal == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.grep(/^[cd]/i)
```

Comment:
Making it hard on yourself!  I like the creativity, before using a Regex, can you think of any other methods you might try?

Reply:
Potentially the use of input.select or input.find_all in order to locate the "c" or "d". 

input.find_all {|animal| animal.start_with?('c', 'd')}

Reply:
Perfect - What you did was fine, just wanted to suggest that regex aren't always the best solution even if they are very powerful (hard to read and maintain).